test/core/security/tls_security_connector_test.cc

```diff
@@ -376,6 +376,133 @@ TEST_F(TlsSecurityConnectorTest, TlsCheckHostNameFail) {
   GRPC_ERROR_UNREF(error);
 }
 
+TEST_F(TlsSecurityConnectorTest,
+       CompareChannelSecurityConnectorSucceedsOnSameCredentials) {
+  grpc_core::RefCountedPtr<grpc_tls_certificate_distributor> distributor =
+      grpc_core::MakeRefCounted<grpc_tls_certificate_distributor>();
+  distributor->SetKeyMaterials(kRootCertName, root_cert_0_, absl::nullopt);
+  grpc_core::RefCountedPtr<::grpc_tls_certificate_provider> provider =
+      grpc_core::MakeRefCounted<TlsTestCertificateProvider>(distributor);
+  auto options = grpc_core::MakeRefCounted<grpc_tls_credentials_options>();
+  options->set_certificate_provider(provider);
+  options->set_watch_root_cert(true);
+  options->set_root_cert_name(kRootCertName);
+  grpc_core::RefCountedPtr<TlsCredentials> credential =
+      grpc_core::MakeRefCounted<TlsCredentials>(options);
+  grpc_core::RefCountedPtr<grpc_channel_security_connector> connector =
+      credential->create_security_connector(nullptr, kTargetName, nullptr,
+                                            nullptr);
+  grpc_core::RefCountedPtr<grpc_channel_security_connector> other_connector =
+      credential->create_security_connector(nullptr, kTargetName, nullptr,
+                                            nullptr);
+  // Comparing the equality of security connectors generated from the same
+  // channel credentials with same settings should succeed.
+  EXPECT_EQ(connector->cmp(other_connector.get()), 0);
+}
+
+TEST_F(TlsSecurityConnectorTest,
+       CompareChannelSecurityConnectorSucceedsOnDifferentChannelArgs) {
```

Comment:
This test isn't really useful, since the security connector doesn't store channel args.  The fact that the args are passed to `create_security_connector()` is just to give the credentials object a chance to modify the channel args for the subchannel; it really has nothing to do with the resulting security connector.